lib/cache.gd

```diff
@@ -0,0 +1,55 @@
+#############################################################################
+##
+#W  cache.gd                     GAP library                 Chris Jefferson
+##
+##
+#Y  Copyright (C) 2017 University of St Andrews, Scotland
+##
+##  This file defines various types of caching data structures
+##
+
+#############################################################################
+##
+#F  MemoizePosIntFunction(<function> (,<defaults> ) )
+##
+##  <#GAPDoc Label="MemoizePosIntFunction">
+##  <ManSection>
+##  <Func Name="MemoizePosIntFunction" Arg='function (,defaults)'/>
+##
+##  <Description>
+##  <Ref Func="MemoizePosIntFunction"/> returns a function which behaves the same
+##  as <A>function</A>, except it caches generated results to avoid recalling
+##  the function.
+##  <P/>
+##  The returned function will only accept positive integers.
+##  <P/>
+##  This function does not promise to never call <A>function</A> more than
+##  once for any input -- values may be removed if the cache gets too large,
+##  or GAP chooses to flush all caches, or if multiple threads try to calculate
+##  the same value simultaneously.
+##  <P/>
+##  <A>defaults</A>, if given, gives a list of previously computed values which
```

Comment:
"... if given, gives ..." confused me. Perhaps: "if given, describes" or "is expected to be"?